okhttp/src/main/kotlin/okhttp3/internal/publicsuffix/PublicSuffixDatabase.kt

```diff
@@ -338,7 +239,7 @@ class PublicSuffixDatabase internal constructor(
             low = mid + end + 1
           } else {
             // Found a match.
-            match = String(this, mid, publicSuffixLength)
+            match = this.substring(mid, mid + publicSuffixLength).string(Charsets.UTF_8)
```

Comment:
Seems like `string` should have an overload which takes start/end indices to avoid this double copy. Or we should have added [`ByteString.slice`](https://github.com/square/okio/issues/668)!

Reply:
Will raise a PR.